src/frontend/screens/Observation/Buttons.tsx

```diff
@@ -117,36 +113,47 @@ export const ButtonFields = ({
     ]);
   }
 
-  async function handlePressShare() {
+  async function fetchFreshUrls() {
```

Comment:
I think this can be simplified a bit. 2 main things. if you await`Promise.all`, it resolves (or rejects) all the promises inside the array. So it is redundant (and in certain edgecases lead to unwanted behaviour) to await all the objects inside the array. 

Secondly you can prefilter the array using the functions you created to determine if it is type audio or photo before you put them in the `promise.all`. That way you dont have to do a bunch of checks afterwards for null/undefined as well as type photo and audio

Here is what I would suggest

```ts

//defined outside the react component:
function isPhotoOrAudio(
  attachment: Attachment,
): attachment is Attachment & {type: 'photo' | 'audio'} {
  return isSavedPhoto(attachment) || isAudioAttachment(attachment);
}

// inside the react component:
async function fetchFreshUrls() {
    const {attachments} = observation;

    if (attachments.length === 0) {
      return [];
    }

    return await Promise.all(
      attachments.filter(isPhotoOrAudio).map(attachment => {
        return projectApi.$blobs.getUrl({
          driveId: attachment.driveDiscoveryId,
          name: attachment.name,
          type: attachment.type,
          variant: 'original',
        });
      }),
    );
  }

Reply:
[9af300f](https://github.com/digidem/comapeo-mobile/pull/847/commits/9af300f79d54f28a85d7487d12fb983f71421c2f)